config/config.go

```diff
@@ -1123,6 +1158,20 @@ func migrateConfig(v *viper.Viper) {
 	}
 }
 
+func migrateConfigCompression(v *viper.Viper) {
+	oldField := "enable_gzip"
+	newField := "compression.response.enabled"
+	if v.IsSet(oldField) {
+		oldConfig := v.GetBool(oldField)
+		if v.IsSet(newField) {
+			glog.Warningf("using %s and ignoring deprecated %s", newField, oldField)
+		} else {
+			glog.Warningf("%s is deprecated and should be changed to %s", oldField, newField)
+			v.Set(newField, oldConfig)
+		}
+	}
+}
```

Comment:
Does setting `SetDefault` cause the `IsSet` to always be true?

Reply:
Yes, setting `SetDefault` causes `IsSet` to be true. if we call `SetDefault` before `migrateConfigCompression`, we will always end up executing the migration logic because `IsSet` will return true. So in order for the migration logic to be only executed if the old field is explicitly set by the host, we need to set the default values after the migration logic.

Reply:
Please capture that requirement in a comment, if it's not already.

Reply:
Yes, it is already captured [here](https://github.com/prebid/prebid-server/blob/master/config/config.go#L1056).